daffodil-core/src/main/scala/org/apache/daffodil/grammar/AlignedMixin.scala

```diff
@@ -48,13 +51,29 @@ trait AlignedMixin extends GrammarMixin { self: Term =>
 
   requiredEvaluationsIfActivated(hasNoSkipRegions)
 
+  /**
+   * If "manual" this property disables all automatic alignment. The
+   * schema author must use leadingSkip, trailingSkip, or just ensure
+   * all the elements/terms are alignmed based on their length.
+   */
+  lazy val alignmentKind: AlignmentKind = {
+    val prop = findPropertyOption("alignmentKind")
+    prop match {
+      case NotFound(_, _, _) => AlignmentKind.Automatic // default since this is a new property
+      case Found("automatic",_,_,_) => AlignmentKind.Automatic
+      case Found("manual",_,_,_) => AlignmentKind.Manual
+      case _ => Assert.impossible("Per validation, can only be 'automatic' or 'manual'")
+    }
```

Comment:
Much more readable now!  Even covers the impossible case so readers will understand what's happening.

Reply:
You know, that may be the case. I may have over complicated this. This is a regular property, except extension namespace, and default value.